test/e2e/antreapolicy_test.go

```diff
@@ -1568,6 +1562,20 @@ func executeTestsWithData(t *testing.T, testList []*TestCase, data *TestData) {
 			for _, p := range step.CustomProbes {
 				doProbe(t, data, p)
 			}
+			log.Debugf("cleaning-up policies created by this test step and sleeping for %v", networkPolicyDelay)
+			if k8sNPCreated {
+				failOnError(k8sUtils.CleanNetworkPolicies(namespaces), t)
+			}
+			if acnpCreated {
+				failOnError(k8sUtils.CleanACNPs(), t)
+			}
+			if anpCreated {
+				failOnError(k8sUtils.CleanANPs(namespaces), t)
+			}
+			if cgCreated {
+				failOnError(k8sUtils.CleanCGs(), t)
+			}
```

Comment:
I feel like the `if` statements are not necessary and we can unconditionally call the `Clean*` functions, even if it means a `List` API request to the apiserver

Reply:
The purpose of adding the `if` statements are ensuring that only the resources created by the specific testcase is cleaned up. If you take the `TestGroupDefaultDENY` for example, we want to ensure the K8s default deny netpol are only cleaned up after all the testcases in the group is executed. This 'hold off cleanup' however does only apply to K8s netpol as of now. I can remove some of the `if` statements here.

Reply:
can't we iterate over `step.Policies` and delete these policies then?
it would be a counterpart to the `applyPolicies` function: same structure, but would delete the exact set of policies instead of creating them

Reply:
Make sense. Let me see if we could change it that way.